### PR TITLE
Y24-286-1 - Fix RVI BCL Library prep requests

### PR DESCRIPTION
**Pull request by BenTopping**
Part 1 of https://github.com/sanger/sequencescape/issues/4306

#### Changes proposed in this pull request

- Updates transfer request to start sibiling requests when going from pending to processed_1

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: spec/models/state_changer/standard_plate_spec.rb [hunk 1]
```diff
@@ -21,12 +21,25 @@
...
-        expect { state_changer.update_labware_state }.to change(BroadcastEvent::LibraryStart, :count).by(1)
-        expect(library_requests).to all(be_started)
+        it 'starts the requests', :aggregate_failures do
+          # Requests are started and we create one event per order.
+          expect { state_changer.update_labware_state }.to change(BroadcastEvent::LibraryStart, :count).by(1)
+          expect(library_requests).to all(be_started)
+        end
+      end
+
+      # RVI BCL uses the processed_1 state as the first state for the the intial plate
```
Comment (andrewsparkes) [inline spec/models/state_changer/standard_plate_spec.rb:34]: typo intial -> initial

Comment (andrewsparkes) [reply spec/models/state_changer/standard_plate_spec.rb:34]: My first question would be why doesn't RVI used the 'started' state as the first state? So it sticks to the same convention as others. I forget the details of this pipeline...

My second question is, what happens if a pipeline uses both 'started' and 'processed_1' states? It doesn't trigger LibraryStart twice does it?

File: app/models/transfer_request.rb [hunk 2]
```diff
@@ -80,7 +80,7 @@ class TransferRequest < ApplicationRecord # rubocop:todo Metrics/ClassLength
     end
 
     event :process_1 do
-      transitions to: :processed_1, from: [:pending]
+      transitions to: :processed_1, from: [:pending], after: :on_started
```
Comment (andrewsparkes) [inline app/models/transfer_request.rb:83]: what does that change do? what is on_started ?

**Review by andrewsparkes — COMMENTED** [review]
I had questions!

File: spec/models/state_changer/standard_plate_spec.rb (hunk 1 above)
Comment (BenTopping) [reply spec/models/state_changer/standard_plate_spec.rb:34]: Good question. The RVI RT plate (the second plate in the pipeline) has 4 operations on it which require bed verifications so it needs to move through a lot of states. The best way to do this was to move it through the various processed states (I think we added processed_3 and processed_4 for this).

In hindsight moving to 'started' first and then the other states might have made more sense, but started is not a required first state. The on_started method is also called when moving from 'pending' to 'passed' which most pipelines do every plate transition.

No pipeline has both started and processed_1 but if they did nothing would happen, once the requests are started they cannot be started again.

**Pull request merged**